src/Domain/Donation.php

```diff
@@ -1806,4 +1810,49 @@ public function isRegularGiving(): bool
     {
         return $this->mandate !== null;
     }
+
+    /**
+     * @throws CannotRemoveGiftAid
+     */
+    public function removeGiftAid(\DateTimeImmutable $at): void
+    {
+        // This function will be called via the Salesforce UI, so its safe to assume the donation has an SF ID by now.
+        $salesforceId = $this->getSalesforceId();
+
+        if ($this->tbgGiftAidRequestQueuedAt !== null) {
+            $formattedQueuedDate = $this->tbgGiftAidRequestQueuedAt->format('Y-m-d H:i');
+
+            throw new CannotRemoveGiftAid(
+                "Cannot remove gift aid from donation {$salesforceId}, request already " .
+                "queued to send to HMRC at {$formattedQueuedDate}"
+            );
+        }
+
+        if ($this->giftAidRemovedAt !== null) {
+            $formattedQueuedDate = $this->giftAidRemovedAt->format('Y-m-d H:i');
+
+            // This function will be called via the Salesforce UI, so its safe to assume the donation has an SF ID by now.
+            $salesforceId = $this->getSalesforceId();
+
+            throw new CannotRemoveGiftAid(
+                "Cannot remove gift aid from donation {$salesforceId}, gift aid " .
+                "already removed at {$formattedQueuedDate}"
+            );
+        }
+
+        if (!$this->giftAid && !$this->tipGiftAid) {
+            throw new CannotRemoveGiftAid(
+                "Cannot remove gift aid from donation {$salesforceId}, gift aid " .
+                "was never claimed"
```

Comment:
```suggestion
                "eligibility was never reported by donor"
```
Just trying to disambiguate as I think "claimed" could imply the other meaning.

Reply:
> 
```suggestion
                "was not requested by donor"
```

Reply:
Thanks, I made a slightly different change to the message.